
[issue 1-8 are part of github-only]

issue 9:
    when you add new blocks without rebuilding, BlockItemGenerator creates the item images, but fails to create
        Item-instances
    this issue occurs only in the first run with the new block

issue 10:
    occurs since: upgrade to python 3.8, release date of it

    in python 3.8.1, a lot of exceptions of the following style are printed out when closing the window:
        Exception ignored in: <function WeakMethod.__new__.<locals>._cb at 0x0F426D18>
        Traceback (most recent call last):
          File "<path to your python>\lib\weakref.py", line 58, in _cb
          File "<path to your python>\lib\site-packages\pyglet\event.py", line 364, in _remove_handler
          File "<path to your python>\lib\weakref.py", line 74, in __eq__
        TypeError: isinstance() arg 2 must be a type or tuple of types

    possible fix: implement overwriting file for pyglet/event.py to fix this issue from pyglet side

issue 12:
    occurs since: properly since chat was changed

    when pressing ENTER in start menu, an message called "[CHAT] " is printed out for no reason

    possible fix: check chat system for some non-state related linking
    possible cause: back in the past, chat keys were input twice. In this time, things may have changed

issue 14:
    occurs since: unknown, found on 29th of January, 2020

    game crash in BlockItemGenerator after some images were failed to generate:
         Traceback (most recent call last):
          File "C:/Users/baulu/PycharmProjects/core/__main__.py", line 100, in <module>
            main()
          File "C:/Users/baulu/PycharmProjects/core/__main__.py", line 89, in main
            run()
          File "C:/Users/baulu/PycharmProjects/core/__main__.py", line 83, in run
            pyglet.app.run()
          File "C:\Python\core\3.8.1\lib\site-packages\pyglet\app\__init__.py", line 144, in run
            event_loop.run()
          File "C:\Python\core\3.8.1\lib\site-packages\pyglet\app\base.py", line 175, in run
            self._run()
          File "C:\Python\core\3.8.1\lib\site-packages\pyglet\app\base.py", line 187, in _run
            timeout = self.idle()
          File "C:\Python\core\3.8.1\lib\site-packages\pyglet\app\base.py", line 308, in idle
            redraw_all = self.clock.call_scheduled_functions(dt)
          File "C:\Python\core\3.8.1\lib\site-packages\pyglet\clock.py", line 333, in call_scheduled_functions
            item.func(now - item.last_ts, *item.args, **item.kwargs)
          File "C:\Users\baulu\PycharmProjects\core\event\TickHandler.py", line 41, in tick
            result = function(*args, **kwargs)
          File "C:\Users\baulu\PycharmProjects\core\state\StateBlockItemGenerator.py", line 144, in take_image
            self._error_counter(image, blockname)
          File "C:\Users\baulu\PycharmProjects\core\state\StateBlockItemGenerator.py", line 160, in _error_counter
            self.generate_item(blockname, file)
          File "C:\Users\baulu\PycharmProjects\core\state\StateBlockItemGenerator.py", line 176, in generate_item
            if type(block) != str: block.modify_block_item(obj)
        AttributeError: 'NoneType' object has no attribute 'modify_block_item'

issue 16:
    bug: random ticks are never called
    since: random ticks were disabled
    possible fix: implement an call_in_active_tick-method to TickHandler & use it for random ticks

issue 17:
    bug: when entering special html codes into chat, they are used rather than the blank string
    since: chat is using html label
    possible fix: rewrite chat to exclude these characters

issue 21:
    bug: loading hangs in BlockItemFactory when leaving window during working
    possible fix strategy: use debug mode of running to reproduce

issue 22:
    bug: on an tower of sand blocks if the under block is removed, it updates not the other blocks. If the block above
         is removed, it all updates following bug 23

issue 23:
    bug: when sand is falling, it falls more than 1 block at a time, see issue 22 for reproducing

issue 24:
    bug: hit-box of blocks are at the corners not working [see fogleman #104]
    possible fix: rework whole hit-box system

issue 25:
    bug: crafting does not work for e.g. oak log
    possible strategy for solving: search where recipe checks fails

issue 26:
    bug: when for some reason, exceptions in removing directories fails, it does NOT stop
    since: removing folders in try-except block
    fix: add some counter, crash if an value is reached

issue 28:
    bug: /reload does not update visible state of blocks
    since: unknown, found after fixing issue 19
    fix: debug what is wrong

issue 33:
    bug: mouse middle click on blocks gives always the block, not only if not existent in inventory
    since: unknown
    fix: see Player.add_to_free_place-method

issue 34:
    bug: jumping from downwards into an block, you get damage
    since: block damage was introduced
    fix: rewrite collision system

issue 35:
    bug: going inside an non-full block will give you damage
    since: block damage was introduced
    fix: check for solid block

issue 36:
    bug: at some corners of blocks, HitBox fails in some cases to select the right block
    since: HitBox implementation

issue 39:
    bug: leaves are handled solid
    fix: check where is decided if show block -> check if is_solid_face is used correctly